catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/PdfInputTransformer.java

```diff
@@ -73,8 +123,71 @@ public Metacard transform(InputStream input, String id)
         }
     }
 
+    private Metacard transformWithExtractors(InputStream input, String id)
+            throws IOException, CatalogTransformerException {
+        try (FileBackedOutputStream fbos = new FileBackedOutputStream(1000000)) {
+            try {
+                IOUtils.copy(input, fbos);
+            } catch (IOException e) {
+                throw new CatalogTransformerException("Could not copy bytes of content message.",
+                        e);
+            }
+
+            String plainText = null;
+            try (InputStream isCopy = fbos.asByteSource()
+                    .openStream()) {
+                Parser parser = new AutoDetectParser();
+                ContentHandler contentHandler = new ToTextContentHandler();
+                TikaMetadataExtractor tikaMetadataExtractor = new TikaMetadataExtractor(parser,
+                        contentHandler);
+                tikaMetadataExtractor.parseMetadata(isCopy, new ParseContext());
+                plainText = contentHandler.toString();
```

Comment:
It would only be a vulnerability if it failed open for some reason. We still are going to default to system-high markings if something comes into the system without any markings. It could still blow up the heap though, which is a problem.